subprojects/core/src/main/java/org/gradle/internal/classpath/CombinedFileTransformer.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.classpath;
+
+import org.gradle.internal.snapshot.FileSystemLocationSnapshot;
+
+import java.io.File;
+import java.util.List;
+
+class CombinedFileTransformer implements ClasspathFileTransformer {
```

Comment:
💄 The `Combined` prefix doesn't convey much information to me, I was expecting something `Agent` something. The class could also use a small javadoc sentence about its responsibility of computing a `TransformedClassPath` so it's picked-up by the agent later.